src/main/java/ch/njol/skript/expressions/ExprLowestHighestSolidBlock.java

```diff
@@ -0,0 +1,89 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Lowest/Highest Solid Block")
+@Description("An expression to obtain the lowest or highest solid block at a location.")
+@Examples({
+	"teleport the player to the block above the highest block at the player",
+	"set the highest solid block at the player's location to the lowest non-air block at the player's location"
+})
+@Since("2.2-dev34, INSERT VERSION (lowest solid block)")
+public class ExprLowestHighestSolidBlock extends SimplePropertyExpression<Location, Block> {
+
+	static {
+		Skript.registerExpression(ExprLowestHighestSolidBlock.class, Block.class, ExpressionType.COMBINED,
+			"[the] (highest|:lowest) [solid|non-air] block (at|of) %locations%"
```

Comment:
Non-air doesn't mean solid, so they shouldn't be treated as the same.

Reply:
Hmm it's a breaking change but this is true lol (not sure if highest point method has support for this or not)

Reply:
[World#getHighestBlockAt(int x, int z)](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/World.html#getHighestBlockAt(int,int)) always returns the highest solid and doesn't support anything else on itself. Probably will have to manually write that code for non-air if you want to do that. Might as well just remove non-air from the pattern and force people to write their own loop for this, since I can't imagine many people using it.

Reply:
I'm going to take this approach of removing support for it.